data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/file/FileWatcher.java

```diff
@@ -62,18 +65,27 @@ public class FileWatcher implements AutoCloseable {
      * @param contractConsumer updates receiver.
      * @param file             file to watch
      */
-    public FileWatcher(File file, Consumer<DataPlaneContract.Contract> contractConsumer) {
+    public FileWatcher(File file) {
         Objects.requireNonNull(file, "provide file");
-        Objects.requireNonNull(contractConsumer, "provide consumer");
 
-        this.contractConsumer = contractConsumer;
+        this.contractConsumer = null;
+        this.triggerFunction = null;
         this.toWatch = file.getAbsoluteFile();
         this.lastContract = -1;
     }
 
+    public void setContractConsumer(Consumer<DataPlaneContract.Contract> contractConsumer) {
+        this.contractConsumer = contractConsumer;
+    }
+
+    public void setTriggerFunction(Runnable triggerFunction) {
+        this.triggerFunction = triggerFunction;
+    }
```

Comment:
Can we use the constructor rather than set methods? I'm pretty sure that set methods make the class not thread safe anymore because calling them will have an undefined behavior / data race

Reply:
Also if we want to make this class generic it should have no awareness of `DataPlaneContract.Contract`

Reply:
We should abstract consumer/trigger away, they are basically the same think